mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSModelAsNode.kt

```diff
@@ -54,10 +57,26 @@ data class MPSModelAsNode(val model: SModel) : IDeprecatedNodeDefaults {
     }
 
     override val allChildren: Iterable<INode>
-        get() = model.rootNodes.map { MPSNode(it) }
+        get() {
+            val childLinks = listOf(builtinModel.rootNodes, builtinModel.modelImports, builtinModel.usedLanguages)
+            return childLinks.flatMap { getChildren(it) }
+        }
 
     override fun removeChild(child: INode) {
-        TODO("Not yet implemented")
+        val link = child.getContainmentLink() ?: throw RuntimeException("ContainmentLink not found for node $child")
+        if (link.conformsTo(builtinModel.usedLanguages)) {
+            removeUsedLanguage(child)
+        }
+        throw UnsupportedOperationException("Cannot remove child in link $link")
```

Comment:
Is there a reason why only languages can be removed and nothing else? If so, please document it.